bql/planner/planner_test.go

```diff
@@ -491,6 +480,258 @@ func TestPlannerQuery(t *testing.T) {
+	                   ?s "connected_to"@[] ?o }
+                into ?dest
+	        from ?src
+	        where {?s "met"@[] ?o.
+		       ?s "met_at"@[?t] ?o};`
+	p, err := grammar.NewParser(grammar.SemanticBQL())
+	if err != nil {
+		t.Errorf("grammar.NewParser: should have produced a valid BQL parser, %v", err)
+	}
+	s, ctx := memory.NewStore(), context.Background()
+	populateStoreWithTriples(s, ctx, "?src", constructTestSrcTriples, t)
+	populateStoreWithTriples(s, ctx, "?dest", "", t)
+
+	st := &semantic.Statement{}
+	if err := p.Parse(grammar.NewLLk(bql, 1), st); err != nil {
+		t.Errorf("Parser.consume: failed to parse query %q with error %v", bql, err)
+	}
+	plnr, err := New(ctx, s, st, 0, nil)
+	if err != nil {
+		t.Errorf("planner.New failed to create a valid query plan with error %v", err)
+	}
+	_, err = plnr.Execute(ctx)
+	if err != nil {
+		t.Errorf("planner.Execute failed for query %q with error %v", bql, err)
+	}
+
+	g, err := s.Graph(ctx, "?dest")
+	if err != nil {
+		t.Errorf("memory.DefaultStore.Graph(%q) should have not fail with error %v", "?test", err)
+	}
+
+	ts := make(chan *triple.Triple)
+	go func() {
+		if err := g.Triples(ctx, storage.DefaultLookup, ts); err != nil {
+			t.Error(err)
+		}
+	}()
+
+	bnm := make(map[string]map[string]bool)
+	bns := make(map[string]string)
+	bna := map[string]bool{
+		"/_<b1>": true,
+		"/_<b2>": true,
+	}
+	dtm := map[string]bool{
+		fmt.Sprintf("%s\t%s\t%s", `/person<A>`, `"connected_to"@[]`, `/person<B>`):                                 false,
+		fmt.Sprintf("%s\t%s\t%s", `/person<B>`, `"connected_to"@[]`, `/person<C>`):                                 false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b1>`, `"_subject"@[2016-04-10T04:25:00Z]`, `/person<A>`):                     false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b1>`, `"_predicate"@[2016-04-10T04:25:00Z]`, `"met"@[2016-04-10T04:25:00Z]`): false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b1>`, `"_object"@[2016-04-10T04:25:00Z]`, `/person<B>`):                      false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b1>`, `"location"@[]`, `/city<New York>`):                                    false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b1>`, `"outcome"@[]`, `"good"^^type:text`):                                   false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b2>`, `"_subject"@[2016-04-10T04:25:00Z]`, `/person<B>`):                     false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b2>`, `"_predicate"@[2016-04-10T04:25:00Z]`, `"met"@[2016-04-10T04:25:00Z]`): false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b2>`, `"_object"@[2016-04-10T04:25:00Z]`, `/person<C>`):                      false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b2>`, `"location"@[]`, `/city<New York>`):                                    false,
+		fmt.Sprintf("%s\t%s\t%s", `/_<b2>`, `"outcome"@[]`, `"good"^^type:text`):                                   false,
+	}
+
+	// First, we map each blank node generated to a potential blank node placeholder (such as b1 or b2.)
```

Comment:
Yep. It could be avoided if we can mock uuid.NewRandom() to produce deterministic output, though currently this setup is pretty verbose and informative when the set of stored triples are not as expected. Do you think it worth exploring the mocking route?